test/tests-rest-spread-properties.js

```diff
@@ -0,0 +1,825 @@
+            "method": false,
+            "shorthand": true,
+            "computed": false,
+            "key": {
+              "type": "Identifier",
+              "start": 7,
+              "end": 8,
+              "name": "b"
+            },
+            "kind": "init",
+            "value": {
+              "type": "Identifier",
+              "start": 7,
+              "end": 8,
+              "name": "b"
+            }
+          },
+          {
+            "type": "Property",
+            "start": 9,
+            "end": 10,
+            "method": false,
+            "shorthand": true,
+            "computed": false,
+            "key": {
+              "type": "Identifier",
+              "start": 9,
+              "end": 10,
+              "name": "c"
+            },
+            "kind": "init",
+            "value": {
+              "type": "Identifier",
+              "start": 9,
+              "end": 10,
+              "name": "c"
+            }
+          }
+        ]
+      }
+    }
+  ],
+  "sourceType": "script"
+}, { ecmaVersion: 9 })
+test("({...(a,b),c})", {
+  "type": "Program",
+  "start": 0,
+  "end": 14,
+  "body": [
+    {
+      "type": "ExpressionStatement",
+      "start": 0,
+      "end": 14,
+      "expression": {
+        "type": "ObjectExpression",
+        "start": 1,
+        "end": 13,
+        "properties": [
+          {
+            "type": "SpreadElement",
+            "start": 2,
+            "end": 10,
+            "argument": {
+              "type": "SequenceExpression",
+              "start": 6,
+              "end": 9,
+              "expressions": [
+                {
+                  "type": "Identifier",
+                  "start": 6,
+                  "end": 7,
+                  "name": "a"
+                },
+                {
+                  "type": "Identifier",
+                  "start": 8,
+                  "end": 9,
+                  "name": "b"
+                }
+              ]
+            }
+          },
+          {
+            "type": "Property",
+            "start": 11,
+            "end": 12,
+            "method": false,
+            "shorthand": true,
+            "computed": false,
+            "key": {
+              "type": "Identifier",
+              "start": 11,
+              "end": 12,
+              "name": "c"
+            },
+            "kind": "init",
+            "value": {
+              "type": "Identifier",
+              "start": 11,
+              "end": 12,
+              "name": "c"
+            }
+          }
+        ]
+      }
+    }
+  ],
+  "sourceType": "script"
+}, { ecmaVersion: 9 })
+
+testFail("({...})", "Unexpected token (1:5)", { ecmaVersion: 9 })
+testFail("({...obj})", "Unexpected token (1:2)", { ecmaVersion: 8 })
+
+//------------------------------------------------------------------------------
+// Rest Properties
+//------------------------------------------------------------------------------
+
+test("({...obj} = foo)", {
```

Comment:
This and below are destructuring assignments.

Reply:
Sorry, I didn't mean assignments, I meant declarations.

Reply:
Ah, I see. I will add it.